filter/partial_clone_test.go

```diff
@@ -0,0 +1,139 @@
+package filter
+
+import (
+	"testing"
+
+	"github.com/AdRoll/baker"
+)
+
+func TestNewPartialClone(t *testing.T) {
+	fieldByName := func(fname string) (baker.FieldIndex, bool) {
+		switch fname {
+		case "field0":
+			return 0, true
+		case "field1":
+			return 1, true
+		case "field2":
+			return 2, true
+		default:
+			return -1, false
+		}
+	}
+
+	tests := []struct {
+		name         string
+		confFields   []string
+		wantFieldIdx []baker.FieldIndex
+		wantErr      bool
+	}{
+		{
+			name:         "base conf",
+			confFields:   []string{"field0", "field2"},
+			wantFieldIdx: []baker.FieldIndex{0, 2},
+		},
+		{
+			name:         "empty conf",
+			confFields:   []string{},
+			wantFieldIdx: []baker.FieldIndex{},
+			wantErr:      true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			params := baker.FilterParams{
+				ComponentParams: baker.ComponentParams{
+					FieldByName: fieldByName,
+					DecodedConfig: &PartialCloneConfig{
+						Fields: tt.confFields,
+					},
+				},
+			}
+
+			filter, err := NewPartialClone(params)
+			if (err == nil) == tt.wantErr {
+				t.Fatalf("err: %v, wantErr: %t", err, tt.wantErr)
+			}
+
+			if tt.wantErr {
+				return
+			}
+
+			if len(filter.(*PartialClone).fieldIdx) != len(tt.confFields) {
+				t.Errorf("fields len, want: %d, got: %d", len(tt.confFields), len(filter.(*PartialClone).fieldIdx))
+			}
+
+			for _, idx := range tt.wantFieldIdx {
+				got := false
+				for _, fieldIdx := range filter.(*PartialClone).fieldIdx {
+					if fieldIdx == idx {
+						got = true
+						break
+					}
+				}
+				if !got {
+					t.Errorf("can't find field %d in fieldIdx", idx)
+				}
+			}
+		})
+	}
+}
+
+func TestPartialClone_Process(t *testing.T) {
+	fieldByName := func(fname string) (baker.FieldIndex, bool) {
+		switch fname {
+		case "field0":
+			return 0, true
+		case "field1":
+			return 1, true
+		case "field2":
+			return 2, true
+		default:
+			return -1, false
+		}
+	}
+
+	tests := []struct {
+		name       string
+		record     string
+		wantRecord string
+	}{
+		{
+			name:       "base",
+			record:     "a,foo,bar",
+			wantRecord: "a,,bar,,",
```

Comment:
I think it would probably be better if we weren't comparing the serialized version of the processed record. These trailing `,` are due to a bug that still needs to be fixed.
Why not call a serie of record.Get and Set on it?

Reply:
the idea was to avoid to call extra functions (to avoid adding more hidden logic), but I see the point here. fixed